### ISSUE TITLE
[studio] Studio search filters do not return expected searches

### ISSUE THREAD

**Issue by AndyReiter** [craftercms/craftercms#5793]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] The issue is in the latest released 4.0.x
- [ ] The issue is in the latest released 3.1.x

### Describe the issue

When filtering a search based on content types and other filters in studio, the returned items are not correct in accordance with the filters. When the user filters based on content-type AND Last Edited Date, scripts and other incorrect items are returned in the search.

### Steps to reproduce

Steps:

1. Create a site based on Editorial
2. Create a new article and save it
3. In the top right of the studio screen, click the magnifying glass to initiate a search
4. On the left-hand side of the search dialogue, enable the following filters: Content-Type - "Article" and Sort By - "Last Edited Date"
5. Click Apply
6. See that there are items returned that do not follow the filters.

### Relevant log output

```shell
The query being made: 

{"bool":{"filter":[{"bool":{"must":[{"bool":{"should":[{"match":{"content-type":{"query":"/page/article","operator":"OR","prefix_length":0,"max_expansions":50,"fuzzy_transpositions":true,"lenient":false,"zero_terms_query":"NONE","auto_generate_synonyms_phrase_query":true,"boost":1.0}}}],"adjust_pure_negative":true,"boost":1.0}}],"adjust_pure_negative":true,"boost":1.0}}],"adjust_pure_negative":true,"boost":1.0}},"sort":[{"lastEditedOn":{"order":"desc"}}],"aggregations":{"mime-type":{"terms":{"field":"contentType","size":1000,"min_doc_count":1,"shard_min_doc_count":0,"show_term_doc_count_error":false,"order":[{"_count":"desc"},{"_key":"asc"}]}},"size":{"range":{"field":"contentLength","ranges":[{"to":"1024"},{"from":"1024","to":"25600"},{"from":"25600","to":"256000"},{"from":"256000","to":"1048576"},{"from":"1048576","to":"5242880"},{"from":"5242880","to":"1073741824"},{"from":"1073741824"}],"keyed":true}},"last-edit-date":{"date_range":{"field":"lastEditedOn","ranges":[{"key":"today","from":"now-1d","to":"now+1d"},{"key":"in-last-week","from":"now/d-7d","to":"now/d-1d"},{"key":"over-a-month-ago","to":"now/d-30d"},{"key":"over-six-months-ago","to":"now/d-180d"},{"key":"over-a-week-ago","to":"now/d-8d"},{"key":"over-a-year-ago","to":"now/d-365d"}],"keyed":true}},"width":{"range":{"field":"width","ranges":[{"to":"64"},{"from":"64","to":"160"},{"from":"160","to":"480"},{"from":"480","to":"800"},{"from":"800","to":"1280"},{"from":"1280","to":"1600"},{"from":"1600"}],"keyed":true}},"content-type":{"terms":{"field":"content-type","size":1000,"min_doc_count":1,"shard_min_doc_count":0,"show_term_doc_count_error":false,"order":[{"_count":"desc"},{"_key":"asc"}]}},"height":{"range":{"field":"height","ranges":[{"to":"64"},{"from":"64","to":"120"},{"from":"120","to":"320"},{"from":"320","to":"480"},{"from":"480","to":"800"},{"from":"800","to":"1200"},{"from":"1200"}],"keyed":true}}},"highlight":{"fields":{"internal-name":{"fragment_size":50,"number_of_fragments":1},"*_html":{"fragment_size":50,"number_of_fragments":1},"*_txt":{"fragment_size":50,"number_of_fragments":1},"title":{"fragment_size":50,"number_of_fragments":1},"content":{"fragment_size":50,"number_of_fragments":1},"*_t":{"fragment_size":50,"number_of_fragments":1}}}}], id[]
```

### Screenshots and/or videos

![image](https://user-images.githubusercontent.com/42217570/191804776-12f3cb6b-d4be-4ba9-935d-93bb42591007.png)

**Comment by rart** [comment]
Why is this tagged as "ui"? Is there anything wrong with the query?
The "query being made" shared in the description is incomplete (trimmed) and looks nothing like what is issued when following the reproduction steps. Where was that gathered from? Please fix by providing the full query, so we can inspect properly.

Finally, can't reproduce. The query made following the reproduction steps looks like below, which seems fine and produces the right results. Steps/context are missing here.
```json
{
  "query": "",
  "keywords": "",
  "offset": 0,
  "limit": 21,
  "sortBy": "last-edit-date",
  "sortOrder": "desc",
  "filters": {
    "content-type": [
      "/page/article"
    ]
  }
}
```

**Comment by sumerjabri** [comment]
@AndyReiter please provide a video that demonstrates this issue from a clean Editorial. Thanks!

**Comment by AndyReiter** [comment]
I misspoke when writing the ticket, you must create a new article and save it. I updated the steps, also here is a video with a fresh editorial: https://www.loom.com/share/d303bbc3504540838b51d93be4d6603b

**Comment by rart** [comment]
@AndyReiter the query you added in the report is still a mystery where/how it came from/to-be.

@sumerjabri I can reproduce seeing the groovy script showing up when filtering by content type. I don't see anything wrong from the UI side; the query sent looks fine.

```json
{
  "query": "",
  "keywords": "",
  "offset": 0,
  "limit": 21,
  "sortBy": "last-edit-date",
  "sortOrder": "desc",
  "filters": {
    "content-type": [
      "/page/article"
    ]
  }
}
```

**Comment by sumerjabri** [comment]
The backend must not OR the content type with other considerations. Content type can only be ORed with other content types, and then ANDed with other filters.

**Comment by phuongnq** [comment]
This happens because the document `/scripts/components/related-articles.groovy` also has the field `content-type` is `page/article`. See the following screenshot:

![2022-09-30_14-31](https://user-images.githubusercontent.com/2996543/193229604-9e6acfc5-91c6-400f-ae20-a5cccd6cbbbc.png)

While indexing the article, we also check binary files which are referred to by this article. All the metadata from the article will be saved to the new binary indexing metadata, including `content-type`. For this reason, while searching, it will be shown as the query is filtered for any doc that has the field `content-type` is `/page/article`.

![2022-09-30_14-35](https://user-images.githubusercontent.com/2996543/193232472-bb6e5ac5-8582-4d6d-ac2f-dc0199edf494.png)

@sumerjabri isn't this expected for the binary content in authoring? Or do we want to alter how we save the metadata of the binary content?

**Comment by sumerjabri** [comment]
I think I fixed it with https://github.com/craftercms/deployer/commit/b6b04b0c4b29b0501a95af3447922f517b49b8ea

**Comment by AndyReiter** [comment]
Confirmed fixed in 4.0.2-SNAPSHOT. Closing now.
